### PR TITLE
Update auditLog view

### PR DESCRIPTION
**Pull request by VivekVinushanth**
1. Add view to the logs section
2. Popup view modal which has copy, download and close capability
3. Update search placeholder
4. Add columns name to the logs

### CODE REVIEW

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 1]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                            title={ "Copy" }
+                            data-componentid="log-data-viewer-copy-tooltip"
+                        >
+                            <IconButton
+                                size="small"
+                                onClick={ copyCurrentLog }
+                            >
+                                {
+                                    <Icon name="copy"/>
+                                }
```
Comment (DonOmalVindula) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:345]:
```suggestion
                                <Icon name="copy"/>
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 2]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                            title={ "Exit" }
+                            data-componentid="log-data-viewer-exit-tooltip"
+                        >
+                            <IconButton
+                                size="small"
+                                onClick={ handleLogDataViewClose }
+                            >
+                                {
+                                    <Icon name="close"/>
+                                }
```
Comment (DonOmalVindula) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:358]:
```suggestion
                                <Icon name="close"/>
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 3]
```diff
@@ -191,12 +227,23 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                                    <span
+                                        data-testid={ `${ componentId }-${ logObject["id"] }-view-data-button` }
+                                        onClick={ () => handleLogDataView(logObject) }
+                                    >
+                                        <Icon name="eye"/>
+                                        { t("extensions:develop.monitor.filter.viewButton.label") }
+                                    </span>
+                                </td>
+                                <td></td>
+                                <td></td>
```
[the remark this answers is not in the document]
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:240]: I need to add eye icon there. Do we have another way?

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 4]
```diff
@@ -16,16 +16,25 @@
  * under the License.
  */
 
+import Modal from "@mui/material/Modal";
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:19]: Let's import Modal from oxygen-ui instead

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 5]
```diff
@@ -16,16 +16,25 @@
...
+import IconButton from "@oxygen-ui/react/IconButton";
+import Toolbar from "@oxygen-ui/react/Toolbar";
+import Tooltip from "@oxygen-ui/react/Tooltip";
+import Typography from "@oxygen-ui/react/Typography";
 import { IdentifiableComponentInterface } from "@wso2is/core/models";
 import { Popup } from "@wso2is/react-components";
 import { saveAs } from "file-saver";
-import React, { MutableRefObject, ReactElement, UIEventHandler, useEffect, useState } from "react";
+import React, { LazyExoticComponent, MutableRefObject, ReactElement, Suspense, UIEventHandler,
+    lazy,useEffect, useState } from "react";
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:30]:
```suggestion
import React, { 
    LazyExoticComponent,
    MutableRefObject,
    ReactElement,
    Suspense,
    UIEventHandler,
    lazy,
    useEffect,
    useState
} from "react";
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 6]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+        <Box className="log-data-viewer-toolbar-container">
+            <Toolbar variant="dense">
+                <Box>
+                    <Typography>
+                        {
+                            t("extensions:develop.monitor.logView.logDataviewer")
+                        }
+                    </Typography>
+                </Box>
+                <div className="actions">
```
Comment (savindi7) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:320]: We can go with Box instead of divs

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 7]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                    <div className="editor-fullscreen">
+                        <Tooltip
+                            title={ "Download" }
+                            data-componentid="log-data-viewer-download-tooltip"
+                        >
+                            <IconButton
+                                size="small"
+                                onClick={ exportCurrentLog }
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:328]: Instad of defining a new function, let's use the existing exportDataOfLog here, as follows.

```suggestion
                                onClick={ () => exportDataOfLog(currentLog) }
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 8]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                            >
+                                {
+                                    <Icon name="download"/>
+                                }
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:332]: We don't need curly braces around jsx here.

```suggestion
                                <Icon name="download"/>
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:1]: Shall we rename this file to `log-data-viewer-panel` ?

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 9]
```diff
@@ -191,12 +227,23 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                                </td>
+                                <td></td>
+                                <td></td>
                                 <td className="download-data-button">
                                     <span
                                         data-testid={ `${ componentId }-${ logObject["id"] }-download-data-button` }
                                         onClick={ () => exportDataOfLog(logObject) }
                                     >
-                                        <Icon name="download" />
+                                        <Icon name="download"/>
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:246]:
```suggestion
                                        <Icon name="download" />
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 10]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                        {
+                            t("extensions:develop.monitor.logView.logDataviewer")
+                        }
+                    </Typography>
+                </Box>
+                <div className="actions">
+                    <div className="editor-fullscreen">
+                        <Tooltip
+                            title={ "Download" }
+                            data-componentid="log-data-viewer-download-tooltip"
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:324]: Shall we define component ids as follows?
```suggestion
                            data-componentid="{ `${ componentId }-download-tooltip` }
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 11]
```diff
@@ -241,6 +288,82 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                                size="small"
+                                onClick={ copyCurrentLog }
+                            >
+                                {
+                                    <Icon name="copy"/>
+                                }
+                            </IconButton>
+                        </Tooltip>
+                        <Tooltip
+                            title={ "Exit" }
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:349]: Let's i18n this text content

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 12]
```diff
@@ -255,6 +378,33 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
                 { loading && (
                     <LoaderPlaceholder />
                 ) }
+                { !loading && logType === TabIndex.AUDIT_LOGS && (
+                    <div className="log-header">
+                        <div className="log-Recordedtime">
+                            <Typography variant="body1">
+                                {
+                                    t("extensions:develop.monitor.logView.headers.recordedAt")
+                                }
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:387]: Let's fix the formatting in other places as well. 

```suggestion
                                { t("extensions:develop.monitor.logView.headers.recordedAt") }
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 13]
```diff
@@ -303,19 +453,43 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+                                                { log.actionId && (
+                                                    <div className="log-actionid-container">
+                                                        { log.actionId }
+                                                    </div>
+                                                ) }
+                                                { log.action && (
+                                                    <div className="log-actionid-container">
+                                                        { log.action }
+                                                    </div>
+                                                ) }
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:481]: Seems like the same jsx component is duplicated here. Will we be able to use only one?

File: apps/console/src/extensions/i18n/resources/en-US/extensions.ts [hunk 14]
```diff
@@ -2659,7 +2659,7 @@ export const extensions: Extensions = {
                 },
                 searchBar: {
                     placeholderDiagnostic: "Search Logs by Trace ID, Action ID, Client ID, Result Message, or Result Status",
-                    placeholderAudit: "Search Logs by Action, Target ID, Initiator ID, Request ID"
+                    placeholderAudit: "Search Logs by actionId, targetId, initiatorId, traceId"
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/i18n/resources/en-US/extensions.ts:2662]: Is this change intentional? IMO, we should keep the existing text content for better UX.

File: apps/console/src/extensions/i18n/resources/en-US/extensions.ts [hunk 15]
```diff
@@ -2681,7 +2684,13 @@ export const extensions: Extensions = {
...
                 toolTips: {
                     seeMore: "See more"
-                }
+                },
+                headers: {
+                    recordedAt: "Recorded Time",
+                    actionId: "Action",
+                    targetId: "Target"
+                },
+                logDataviewer : "AuditLog Data Viewer"
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/i18n/resources/en-US/extensions.ts:2693]: Will this text be visible in the UI?

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx (hunk 4 above)
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:19]: We don't have it already in Oxygen.

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx (hunk 7 above)
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:328]: I require two versions of export.

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:1]: log-data-viewer-panel is  a part of this infinite scroller.

File: apps/console/src/extensions/i18n/resources/en-US/extensions.ts (hunk 14 above)
Comment (VivekVinushanth) [reply apps/console/src/extensions/i18n/resources/en-US/extensions.ts:2662]: Yes. Intentional. We are aligning with the body of the logs. **Action** have no meaning when the logs only have **actionId**. And can you please explain what it means by `we should keep the existing text content for better UX.`

File: apps/console/src/extensions/i18n/resources/en-US/extensions.ts (hunk 15 above)
Comment (VivekVinushanth) [reply apps/console/src/extensions/i18n/resources/en-US/extensions.ts:2693]: Yes. That's the intention.

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx (hunk 13 above)
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:473]: Conditionally rendered. This has to be improved.

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (VivekVinushanth) [reply apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:19]: It was decided to onboard Modal as a separate effort. 
CC: @brionmario

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.scss [hunk 16]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.scss:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 17]
```diff
@@ -88,13 +112,34 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
         setActiveIndex(tempIndexArr);
     };
 
+    function handleLogDataViewClose(): void {
+        setView(false);
+        setCurrentLog(null);
+    }
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:118]: Please follow the arrow function declarations.
```suggestion
    const handleLogDataViewClose = (): void => {
        setView(false);
        setCurrentLog(null);
    }
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 18]
```diff
@@ -61,6 +83,8 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
 
     const { t } = useTranslation();
     const [ activeIndex, setActiveIndex ] = useState<number[]>([ -1 ]);
+    const [ view, setView ] = useState<boolean>(false);
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:86]: Shall we use more descriptive variable names here?
```suggestion
    const [ isViewModalOpen, setIsViewModalOpen ] = useState<boolean>(false);
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 19]
```diff
@@ -88,13 +112,34 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
         setActiveIndex(tempIndexArr);
     };
 
+    function handleLogDataViewClose(): void {
+        setView(false);
+        setCurrentLog(null);
+    }
+
+    const handleLogDataView = (logObject: InterfaceLogEntry) => {
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:120]: - Please write doc comments for all the newly introduced functions.

- Suggesting a more descriptive name
```suggestion
    const handleLogDataViewOpen = (logObject: InterfaceLogEntry) => {
```

File: modules/theme/src/themes/wso2is/apps/developer-portal.overrides [hunk 20]
```diff
@@ -1567,6 +1567,36 @@ TODO : Move values to separate variables and cleanup styles.
...
+            .log-actionId {
+                margin-right: 30px;
+                padding-left: 8px;
+            }
+
+            .log-targetId {
+                margin-right: 30px;
+            }
+        }
+
```
Comment (JayaShakthi97) [inline modules/theme/src/themes/wso2is/apps/developer-portal.overrides:1597]:
```suggestion
```

File: modules/theme/src/themes/wso2is/apps/developer-portal.overrides [hunk 21]
```diff
@@ -1690,6 +1720,12 @@ TODO : Move values to separate variables and cleanup styles.
         >.log-desc-loader {
 		    width: 70%;
 	    }
+
```
Comment (JayaShakthi97) [inline modules/theme/src/themes/wso2is/apps/developer-portal.overrides:1723]:
```suggestion
```

File: .changeset/tidy-otters-join.md [hunk 22]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (JayaShakthi97) [inline .changeset/tidy-otters-join.md:1]: Let's remove this old changeset file. We can just delete the file and commit.

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 23]
```diff
@@ -88,13 +112,46 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
         setActiveIndex(tempIndexArr);
     };
 
+    /**
+     * Handles the LogData View panel close
+    */
+    const handleLogDataViewClose= (): void => {
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:118]: End comments with a full stop. Please check other places as well.

```suggestion
    /**
     * Handles the LogData View panel close.
    */
    const handleLogDataViewClose = (): void => {
```

File: apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx [hunk 24]
```diff
@@ -88,13 +112,46 @@ const InfiniteScrollContainer = (props: InfiniteScrollContainerPropsInterface):
...
+        setisViewModalOpenl(false);
+        setCurrentLog(null);
+    };
+
+    /**
+     * Handles the LogData View panel open
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/logs/components/infinite-scroll-container.tsx:124]:
```suggestion
     * Handles the LogData View panel open.
```

**Review by JayaShakthi97 — APPROVED** [review]

**Comment by JayaShakthi97** [comment]
Let's fix the latest comments through a followup PR.
